packages/sources/trumatic-matic-exchange-rate/src/transport/crypto.ts

```diff
@@ -0,0 +1,77 @@
+import { TransportDependencies } from '@chainlink/external-adapter-framework/transports'
+import { BaseEndpointTypes, inputParameters } from '../endpoint/crypto'
+import { SubscriptionTransport } from '@chainlink/external-adapter-framework/transports/abstract/subscription'
+import { EndpointContext } from '@chainlink/external-adapter-framework/adapter'
+import { TimestampedAdapterResponse, sleep } from '@chainlink/external-adapter-framework/util'
+import { BigNumber, ethers } from 'ethers'
+import trumaticVaultSharesAbi from '../abi/ITrumaticVaultShares.json'
+
+type ExchangeRateTransportTypes = BaseEndpointTypes
+type RequestParams = typeof inputParameters.validated
+
+type SharePriceResponse = [BigNumber, BigNumber]
+
+class ExchangeRateTransport extends SubscriptionTransport<ExchangeRateTransportTypes> {
```

Comment:
[low priority] Think it would be worth making this into a standardized endpoint/transport type in the framework at a later time?

Reply:
Agree we should see if we can standardise this - one option I’ve been thinking about is extending the ‘view-function’ EA so we can provide a function for processing the Contract output. That would cover the use case for both this and the Frax EA.